chainer/links/normalization/layer_normalization.py

```diff
@@ -0,0 +1,53 @@
+from chainer import initializers
+from chainer import link
+
+from chainer.functions.array import broadcast
+from chainer.functions.math import sqrt
+from chainer.functions.math import square
+from chainer.functions.math import sum
+from chainer.links.connection import scale
+
+
+class LayerNormalization(link.Chain):
+
+    """Layer normalization layer on outputs of linear functions.
+
+    This is a link of "Layer Normalization". This layer
+    normalizes input units by statistics along the second axis,
+    scales and shifts them with :class:`~chainer.links.Scale`.
+
+    Args:
+        size (int): Size of input units.
```

Comment:
Yes, the `Linear` link is an example that implements the parameter shapeholder functionality. The basic idea is that if some parameters have a shape that depends on the size of the input activations, then we should defer initialization of such parameters until the first call of `__call__(x)` because the required size can be extracted from x at that time. This makes the link easier to use because the user does not need to specify this information as an argument to `__init__()`. 

In the case of `Linear` link, `None` can be used to turn on the parameter shapeholder feature. However, that is only for backward compatibility. In the case of your layer normalization link, I think it might be best to remove `size` argument completely from `__init__()`.